analysis/emissionscost.py

```diff
@@ -208,49 +218,70 @@ def shadowcost_tariff(
                     emissions_path=None,
                     flex_capacity=continuous_flexibility,
                     rte=1.0,
-                    min_onsteps=max(int(len(aef) * (system_uptime)), 1),  # assuming daily cycles
+                    min_onsteps=max(int(len(emis) * (system_uptime)), 1),  # assuming daily cycles
                     uptime_equality=uptime_equality,
 )
     flex_cost.build()
     flex_cost.solve(threads = threads)
     # Calculate the net facility load
     net_facility_load_costopt = idxparam_value(flex_cost.model.net_facility_load)
 
-    # calculate the total cost and total emissions using the output power
-    # cost_optimal_cost = tariff@net_facility_load_costopt
-    cost_optimal_cost = flex_cost.model.total_flex_cost_signal()
+    # (1) get the charge dictionary
+    charge_dict = costs.get_charge_dict(
+        startdate_dt, enddate_dt, tariff_data, resolution="1h"
+    )
+
+    # (2) set up consumption data dictionary
+    consumption_data_dict_costopt = {"electric": net_facility_load_costopt}
+
+    cost_optimal_cost, _ = costs.calculate_cost(
+            charge_dict,
+            consumption_data_dict_costopt,
+            resolution="1h",
+            desired_utility="electric",
+            desired_charge_type=None,
+            model=None,
+        )
+
         # TODO: replace the unit conversion using pint
-    cost_optimal_emissions = aef@net_facility_load_costopt* 0.001 # convert from $/kg to $/ton (metric)
+    cost_optimal_emissions = emis@net_facility_load_costopt* 0.001 # convert from $/kg to $/ton (metric)
 
     # Calculate emissions optimal while tracking cost
+
     flex_emissions = flexloadMILP(
         baseload=baseload,
-        costing_type="tariff",
-        cost_signal=tariff_data,
-        costing_path = None,
-        startdate_dt=startdate_dt,
-        enddate_dt=enddate_dt,
-        emissions_signal=aef,
-        emissions_type="aef",
+        emissions_signal=emis,
+        costing_type=None,
+        costing_path=None,
+        emissions_type=emissions_type,
         emissions_path=None,
         flex_capacity=continuous_flexibility,
-        weight_of_cost = 0.0, 
         cost_of_carbon=1.0,
         rte=1.0,
-        min_onsteps=max(int(len(aef) * (system_uptime)), 1),  # assuming daily cycles
-        uptime_equality=uptime_equality,
```

Comment:
I don't think it makes much difference either way, but just curious as to why you removed the flag and hard coded `True` here